cli/cmd/lql.go

```diff
@@ -240,78 +244,132 @@ func inputQueryFromEditor(action string) (query string, err error) {
 	return
 }
 
-func queryErrorCrumbs(query string, err error) error {
-	// not the error we're looking for
-	if !strings.Contains(fmt.Sprintf("%s", err), "unable to translate query blob") {
-		return err
+func parseQuery(s string) (api.NewQuery, error) {
+	var query api.NewQuery
+	var err error
+
+	// valid json
+	if err = json.Unmarshal([]byte(s), &query); err == nil {
+		return query, err
+	}
+	// valid yaml
+	query = api.NewQuery{}
+	err = yaml.Unmarshal([]byte(s), &query)
+	if err == nil && !reflect.DeepEqual(query, api.NewQuery{}) { // empty string unmarshals w/o error
+		return query, nil
+	}
+	// invalid policy
+	return query, queryErrorCrumbs(s)
+}
+
+func parseQueryTime(s string) (time.Time, error) {
+	// empty
+	if s == "" {
+		return time.Time{}, errors.New(fmt.Sprintf("unable to parse time (%s)", s))
+	}
+	// parse time as relative
+	if t, err := lwtime.ParseRelative(s); err == nil {
+		return t, err
+	}
+	// parse time as RFC3339
+	if t, err := time.Parse(time.RFC3339, s); err == nil {
+		return t, err
+	}
+	// parse time as millis
+	if i, err := strconv.ParseInt(s, 10, 64); err == nil {
+		return time.Unix(0, i*int64(time.Millisecond)), err
 	}
+	return time.Time{}, errors.New(fmt.Sprintf("unable to parse time (%s)", s))
+}
+
+func queryErrorCrumbs(q string) error {
 	// smells like json
-	query = strings.ToLower(query)
-	if strings.Contains(query, "start_time_range") ||
-		strings.Contains(query, "end_time_range") ||
-		strings.Contains(query, "lql_id") ||
-		strings.Contains(query, "query_text") {
+	q = strings.TrimSpace(q)
+	if strings.HasPrefix(q, "[") ||
+		strings.HasPrefix(q, "{") {
 
 		return errors.New(`invalid query
 
-It looks like you attempted to submit an LQL query in JSON format.
+It looks like you attempted to submit a query in JSON format.
 Please validate that the JSON is formatted properly and adheres to the following schema:
 
 {
-	"query_text": "MyLQL { source { CloudTrailRawEvents } filter { EVENT_SOURCE = 's3.amazonaws.com' } return { INSERT_ID } }"
-}`)
+    "evaluatorId": "Cloudtrail",
+    "queryId": "MyLQL",
+    "queryText": "MyLQL { source { CloudTrailRawEvents } filter { EVENT_SOURCE = 's3.amazonaws.com' } return { INSERT_ID } }"
+}
+`)
 	}
 	// smells like plain text
 	return errors.New(`invalid query
 	
-It looks like you attempted to submit an LQL query in plain text format.
+It looks like you attempted to submit a query in YAML format.
 Please validate that the text adheres to the following schema:
 
-MyLQL {
-	source {
-		CloudTrailRawEvents
-	}
-	filter {
-		EVENT_SOURCE = 's3.amazonaws.com'
-	}
-	return {
-		INSERT_ID
-	}
-}
+evaluatorId: Cloudtrail
+queryId: MyLQL
+queryText: |-
+  MyLQL {
+      source {
+          CloudTrailRawEvents
+      }
+      filter {
+          EVENT_SOURCE = 's3.amazonaws.com'
+      }
+      return {
+          INSERT_ID
+      }
+  }"
```

Comment:
nit: I don't think we need this `"`
```suggestion
  }
```